routing/missioncontrol.go

```diff
@@ -369,3 +369,215 @@ func (m *MissionControl) GetHistorySnapshot() *MissionControlSnapshot {
 
 	return &snapshot
 }
+
+// reportPaymentFail reports a failed payment to mission control as input for
+// future probability estimates. It returns a bool indicating whether this error
+// is a final error and no further payment attempts need to be made.
+func (m *MissionControl) reportPaymentFail(rt *route.Route,
+	failureSourceIdx int, failure lnwire.FailureMessage) bool {
+
+	var (
+		failureVertex route.Vertex
+	)
+	if failureSourceIdx > 0 {
+		failureVertex = rt.Hops[failureSourceIdx-1].PubKeyBytes
+	} else {
+		failureVertex = rt.SourcePubKey
+	}
+
+	// Always determine chan id ourselves, because a channel
```

Comment:
All the line wrapping in this method looks like it can be updated to fill out each line more.

Reply:
Fixed